ktor-core/src/org/jetbrains/ktor/http/Compression.kt

```diff
@@ -0,0 +1,81 @@
+package org.jetbrains.ktor.http
+
+import org.jetbrains.ktor.application.*
+import java.io.*
+import java.util.zip.*
+
+data class CompressionOptions(var minSize: Long = 0L,
+                              var compressStream: Boolean = true,
+                              var defaultEncoding: String = "deflate",
+                              val compressorRegistry: MutableMap<String, CompressionEncoder> = hashMapOf(),
```

Comment:
this API bugs me. So, the consumer is expected to mutate the map, before calling setupCompression?  Or am I missing something here?